### DISCUSSION TITLE
The solar_system example is wrong... from a physics point of view

### DISCUSSION THREAD

**Discussion by TheQuantumPhysicist** [hecrj/iced#1809] [General]
Not really relevant to the library's functionality, but it's a fun fact that the sign of moon's orbit's curvature around the sun never changes.

![image](https://user-images.githubusercontent.com/8694366/233799437-9c4e5fb1-ce12-4a97-8733-b00b23cf05e0.png)

The library is wonderful and it's irrelevant to the library functionality, it's a fun fact that can be corrected in the solar_system example, and people can be fascinated by it as well :-)

Cheers!

**Comment by TheComamba** [comment]
That __is__ a fun fact! Orbital dynamics are truly weird sometimes.
I am intrigued.

**Comment by TheComamba** [comment]
Ok, I thought a bit about this, and came to the conclusion that it should probably stay "wrong".
This phenomenon ultimately comes from the fact that the sun is rather massive and the earth comparatively close to it. Because Mercury and Venus have no moon, ours is actually the only one that exhibits this behaviour, see:
https://www.mathpages.com/home/kmath405/kmath405.htm
The second formula on that site is the key to why we cannot properly reproduce it:
The earth's orbital radius over period squared must be larger than the moon's orbital radius over period squared,
R/T² > r/t² .
In reality, T/t is approximately 13 (we have about 13 months a year), and R/r is about 400.
If we implemented that latter ratio in our simulation we wouldn't perceive any motion of the moon. At the moment it is implemented at 150/28 = 5.4, so to make the moon-sun orbit convex we'd need to make it's orbital period really low.
I played around with the numbers a bit, I think we could go down to about 370/19 = 19.5. But even then we can allow ourselves a maximum of sqrt(19.5) = 4.4 months per year.
So my (still very volatile and malleable) opinion is that we should stick with our current kind of wrong.
And for my personal sake I'm very glad you posted this. :)

**Comment by TheQuantumPhysicist** [comment]
Thank you for looking into this. I really appreciate your curiosity.

That's fair. From a presentation point of view it's difficult to show the correct behavior.

I wonder whether there's a good place to put some kind of disclaimer/link or similar stating that this is just for display purposes and is incorrect dynamics of the moon. I guess that will have the same effect. I don't know how many picky physicists are out there who will pick on this and repost 😅

But to be clear, I'm happy keeping it as it's, given the constraints. You don't have to change anything.

Cheers!

**Comment by clarkmoody** [comment]
I'd love to see a nice n-body gravity simulator running the solar system in Iced
